src/components/MainButton/MainButton.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Button, Typography } from 'antd';
+import browser from 'webextension-polyfill';
+import { SearchStatus, SearchStatusType } from '@src/lib/internal-types';
+import { ActionTypes } from '@src/platform-message';
+import { buttons as Content } from '@src/content.json'
+import { getMyVisitTab } from '@src/lib/utils/tabs';
+import styles from './MainButton.scss';
+import { AppTestIds } from '../dataTestIds';
+
+const { Text } = Typography;
+
+interface MainButtonProps {
+  searchStatusType: SearchStatusType
+  enabled: boolean
+}
+
+interface SearchMessageProps {
+  searchStatus: SearchStatus
+}
+
+const sendMessageToMyVisitTab = async (action: ActionTypes) => {
+  const maybeMyVisitTab = await getMyVisitTab();
+  if (maybeMyVisitTab) {
+    await browser.tabs.sendMessage(maybeMyVisitTab.id!, { action });
+  }
+}
+
+export const MainButton = ({ searchStatusType, enabled }: MainButtonProps) => {
+  const renderButton = () => {
+    if (searchStatusType === SearchStatusType.Started || searchStatusType === SearchStatusType.Warning) {
+      return <Button onClick={() => sendMessageToMyVisitTab(ActionTypes.StopSearch)} data-testid={AppTestIds.STOP_SEARCH_BUTTON}>
```

Comment:
`renderButton = () => {return ...}` is basically React component. I would maybe try to create a map between predicate and status. Something like:
```
const configurations = {
STOP: {onClick: aFunc, disabled: true | false, content: '...'}
}
```
and doing something like:
```
const {content, ...config} = configuration[status]
return <Button {...config}>{content}</Button>
```
But it might be an overkill :)
Basically this is also simple and readable.